modules/programs/vscode/haskell.nix

```diff
@@ -1,62 +1,33 @@
 { pkgs, config, lib, ... }:
 
-with lib;
+let cfg = config.programs.vscode.haskell;
 
-let
-
-  cfg = config.programs.vscode.haskell;
-
-  defaultHieNixExe = hie-nix.hies + "/bin/hie-wrapper";
-  defaultHieNixExeText =
-    literalExpression ''"''${pkgs.hie-nix.hies}/bin/hie-wrapper"'';
-
-  hie-nix = pkgs.hie-nix or (abort ''
-    vscode.haskell: pkgs.hie-nix missing. Please add an overlay such as:
-    ${exampleOverlay}
-  '');
-
-  exampleOverlay = ''
-    nixpkgs.overlays = [
-      (self: super: { hie-nix = import ~/src/hie-nix {}; })
-    ]
-  '';
-
-in {
+in with lib; {
   options.programs.vscode.haskell = {
     enable = mkEnableOption "Haskell integration for Visual Studio Code";
 
-    hie.enable = mkOption {
-      type = types.bool;
-      default = true;
-      description = "Whether to enable Haskell IDE engine integration.";
-    };
+    useGhcup = mkEnableOption "using GHCup for the Haskell integration";
 
-    hie.executablePath = mkOption {
-      type = types.path;
-      default = defaultHieNixExe;
-      defaultText = defaultHieNixExeText;
-      description = ''
-        The path to the Haskell IDE Engine executable.
-        </para><para>
-        Because hie-nix is not packaged in Nixpkgs, you need to add it as an
-        overlay or set this option. Example overlay configuration:
-        <programlisting language="nix">${exampleOverlay}</programlisting>
-      '';
-      example = literalExpression ''
-        (import ~/src/haskell-ide-engine {}).hies + "/bin/hie-wrapper";
-      '';
+    hlsPackage = mkPackageOption pkgs "Haskell language server" {
+      default = [ "haskell-language-server" ];
     };
+
+    ghcupPackage =
+      mkPackageOption pkgs "GHCup" { default = [ "haskellPackages" "ghcup" ]; };
```

Comment:
I dont get why one would use ghcup when it can be installed via nix

Reply:
GHCup might have versions nixpkgs doesn’t yet have and it installs to the home path so it might be more compatible with some things. All weird issues to have when using home-manager but isn’t it nice to have options?